lib/rdc.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const rdc = (source, destination, options, callback) => {
+  if (typeof callback !== 'function') {
+    if (typeof options !== 'function') throw new Error('ERR_INVALID_CALLBACK');
+    callback = options;
+    options = {};
+  }
+
+  const depth = options.depth;
+  const Transform = options.change.transform;
+  const readStreamOptions = options.change.readStreamOptions;
+  const writeStreamOptions = options.change.writeStreamOptions;
+  const filter = options.filter;
+  const stop = options.stopOnError;
+  const errorLog = options.errorLog;
+  const dereference = options.dereference;
+
+  const error = err => {
+    if (stop) throw new Error(err);
+    else if (errorLog) errorLog.write(JSON.stringify(err));
+    else callback(err);
+  };
+
+  const copyDirectory = target => {
```

Comment:
It should be moved outside function.

Reply:
I can't move this function outside  main because I use ```error()``` fn
```suggestion
  const copyDirectory = target => {
```

Reply:
Then why not change signature of `copyDirectory` to something like this?
```js
  const copyDirectory = (target, options, callback) => {
```
First of all, this shouldn't be named `copyDirectory` since it basically calls `fs.mkdirSync`. Also, it should be asynchronous. And at that point why not just inline it?